lib/sequelize.js

```diff
@@ -1142,18 +1142,18 @@ module.exports = (function() {
           var result = autoCallback(transaction);
           if (!result || !result.then) return reject(new Error('You need to return a promise chain/thenable to the sequelize.transaction() callback'));
 
-          result.then(function (result) {
+          return result.then(function (result) {
             return transaction.commit().then(function () {
               resolve(result);
             });
-          }).then(null, function (err) {
-            transaction.rollback().then(function () {
-              reject(err);
-            }, function () {
-              reject(err);
-            });
           });
-        }).catch(reject);
+        }).catch(function(err) {
```

Comment:
Nevermind that comment, that shouldn't matter with the new approach here actually.